core/src/main/java/org/infinispan/interceptors/distribution/ScatteredDistributionInterceptor.java

```diff
@@ -714,7 +712,7 @@ private Object handlePutMapOnOrigin(InvocationContext ctx, PutMapCommand command
             localEntries.put(key, new MetadataImmortalCacheValue(entry.getValue(), metadata));
             commitSingleEntryIfNewer(ctxEntry, ctx, command);
          } else if (info.primary() == null) {
-            throw AllOwnersLostException.INSTANCE;
+            throw new AllOwnersLostException(cacheTopology.getTopologyId());
```

Comment:
Still, by definition of "all owners" this is what happens. I prefer to throw OTE on events where we detect that current topology id does not match to command topology id, rather than any time we don't know what else to do and hope we could know it later.